Analysis/bin/ShuffleMergeSpectral.cxx

```diff
@@ -0,0 +1,775 @@
+    std::vector<Double_t> datagroup_probs; // probability of getting datagroup
+    std::vector<std::string> datagroup_names; // corresponding datagroup name
+    std::map<TauType, Double_t> ttype_prob; // pair<type, probability> (summed up per all dataset)
+
+    const Double_t pt_max,pt_min;
+    const Double_t eta_max;
+    const Double_t pt_threshold;
+    const Double_t exp_disbalance;
+    size_t n_entries;
+
+    Generator* gen;
+    Discret dist_dataset;
+    Uniform dist_uniform;
+
+};
+
+class ShuffleMergeSpectral {
+public:
+    using Tau = tau_tuple::Tau;
+    using TauType = analysis::TauType;
+    using TauTuple = tau_tuple::TauTuple;
+    using Generator = DataSetProcessor::Generator;
+
+    ShuffleMergeSpectral(const Arguments& _args) :
+        args(_args), pt_bins(ParseBins(args.pt_bins())),
+        eta_bins(ParseBins(args.eta_bins())), tau_ratio(ParseTauTypesR(args.tau_ratio())),
+        max_entries(args.max_entries())
+    {
+      bool verbose = 1;
+		  if(args.n_threads() > 1) ROOT::EnableImplicitMT(args.n_threads());
+
+      const auto disabled_branches_vec = SplitValueList(args.disabled_branches(), false, " ,");
+      disabled_branches.insert(disabled_branches_vec.begin(), disabled_branches_vec.end());
+
+      PrintBins("pt bins", pt_bins);
+      PrintBins("eta bins", eta_bins);
+      const auto all_entries = LoadEntries(args.cfg());
+
+      if(verbose) {
+        for (auto dsc: all_entries) {
+          std::cout << "entry group -> " << dsc.name << std::endl;
+          std::cout << "files: " << dsc.data_files.size() << " "
+                    << "spectrums: " << dsc.spectrum_files.size()
+                    << std::endl;
+        //   for(UInt_t i=0; i<dsc.data_files.size();i++)
+        //     std::cout << dsc.data_files[i] << " "
+        //               << dsc.data_set_names[i] << " "
+        //               << dsc.data_set_names_hashes[i] << std::endl;
+        }
+      }
+
+      if(args.mode() == MergeMode::MergeAll) {
+        entries[args.output()] = all_entries;
+      } else if(args.mode() == MergeMode::MergePerEntry) {
+        for(const auto& entry : all_entries) {
+          const std::string output_name = args.output() + "/" + entry.name + ".root";
+          std::cout << output_name << std::endl;
+          entries[output_name].push_back(entry);
+          }
+      } else {
+        throw exception("Unsupported merging mode = '%1%'.") % args.mode();
+      }
+    }
+
+    void Run()
+    {
+      Generator gen(args.seed());
+
+      for(const auto& e : entries) {
+        const std::string& file_name = e.first;
+        const std::vector<EntryDesc>& entry_list = e.second;
+        std::cout << "Processing:";
+        for(const auto& entry : entry_list)
+          std::cout << ' ' << entry.name;
+        std::cout << "\nOutput: " << file_name << std::endl;
+        auto output_file = root_ext::CreateRootFile(file_name, ROOT::kLZ4, 4);
```

Comment:
Better to have compression level and algorithm as an argument. I'm thinking if it makes sense to switch to the max compression (i.e. LZMA with lvl=9) to minimize the size of the shuffled dataset.